### PR TITLE
♻️ Disable Layers prioritization code

### PR DESCRIPTION
**Pull request by jridgewell**
This splits layers into two parts:

1. Bounding box measurements and scrollbox support
2. A new element prioritization algorithm

To test where the metrics regression is coming from, we're just launching the bounding box measurements in canary. If we can confirm this fixes the metrics, we begin digging into why the prioritization algorithm is affecting the metrics so much.

Additionally, reviewed all code between the Resource measurement and Layers measurement, to ensure the both do the same _exact_ same thing outside of the specific measurements.

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -240,10 +244,6 @@ export class Resources {
       /** @private @const {!./layers-impl.LayoutLayers} */
       this.layers_ = layers;
 
-      layers.onScroll((/* elements */) => {
-        this.schedulePass();
-      });
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply src/service/resources-impl.js]: It's a difference between Resources and Layers. I'm targeting any code that could cause a performance hit.

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -1087,8 +1090,7 @@ export class Resources {
     const box = this.viewport_.getLayoutRect(element);
     const resource = Resource.forElement(element);
     if (box.width != 0 && box.height != 0) {
-      // TODO setRelayoutTop_ is being deprecated.
-      this.setRelayoutTop_(box.top);
+      this.dirtyElement(element);
```
Comment (jeffkaufman) [inline src/service/resources-impl.js]: why is this change in this PR?

Comment (jridgewell) [reply src/service/resources-impl.js]: These change should more-or-less equivalent. The call to `setRelayoutTop_` _could_ receive a different top value (the old one received the uncached box-top and the new one receives the cached box-top).

But if they were ever different, than it was a bug. The relayout top is always compared against cached values, never uncached.

Comment (aghassemi) [reply src/service/resources-impl.js]: @jridgewell what happens if the `relayoutTop` is set by an element that's not in the scroll flow of the document? Could we end up laying out less elements or laying them out later than normal causing the dip?

I see `sticky ad` force calling `collapse` and ending up in here whenever an `amp-ad` inside `amp-sticky-ad` fails to loa, setting `layoutTop` to a fixed element's top which doesn't seem to be relevant to what `relayoutTop` is meant for.

(`amp-flying-carpet` is another one ends up here, surprisingly `attemptCollapse` which normal `amp-ad` calls does not seem to go through this code path).

**Pull request merged**